client/analytics/report/products/table.js

```diff
@@ -248,6 +248,12 @@ class ProductsReportTable extends Component {
 					orderby: query.orderby || 'items_sold',
 					order: query.order || 'desc',
 					extended_info: true,
+					/**
+					 * @TODO: Add this parameter because the filterQuery will be derived from the wrong config
+					 * because it will always look for products config, not categories. The solution is to pass
+					 * down the configs explicitly.
+					 */
+					categories: query.categories,
```

Comment:
Previously in https://github.com/woocommerce/wc-admin/pull/1748 I solved this by mapping endpoints such that filterQuery can use the endpoint to know which config to use according to this map:

https://github.com/woocommerce/wc-admin/blob/31a8e98059f06f1624307f85e12efffba4d04dd0/client/wc-api/reports/utils.js#L20-L37

In a follow up PR, I'd like to pass in the config instead of using this lookup to avoid mapping and any resulting confusion.